tests/integration/pkg/helpers/api_rule.go

```diff
@@ -31,19 +31,36 @@ type apiRuleStatusV2Alpha1 struct {
 // RetryableApiRule wraps any function that modifies or creates an APIRule
 type RetryableApiRule func(k8sClient dynamic.Interface, resources ...unstructured.Unstructured) (*unstructured.Unstructured, error)
 
-// APIRuleWithRetries tries toExecute function and retries with onRetry if APIRule status is "ERROR"
+// APIRuleWithRetries tries toExecute function and retries with onRetry if APIRule status is "ERROR" or "Error"
 func ApplyApiRule(toExecute RetryableApiRule, onRetry RetryableApiRule, k8sClient dynamic.Interface, retryOpts []retry.Option, resources []unstructured.Unstructured) error {
 	res, err := toExecute(k8sClient, resources...)
 	if err != nil {
 		return err
 	}
 
-	apiStatus, err := GetAPIRuleStatus(res)
-	if err != nil {
-		return err
+	var arVersion, code, description string
+
+	if res.Object != nil && res.Object["apiVersion"] != nil {
+		apiVersion := strings.Split(res.Object["apiVersion"].(string), "/")
+
+		if apiVersion[1] == "v1beta1" {
+			arVersion = apiVersion[1]
+			arStatus, err := GetAPIRuleStatus(res)
+			if err != nil {
+				return err
+			}
+			code = arStatus.Status.APIRuleStatus.Code
+		} else if apiVersion[1] == "v2alpha1" {
+			arVersion = apiVersion[1]
+			arStatus, err := GetAPIRuleStatusV2Alpha1(res)
+			if err != nil {
+				return err
+			}
+			code = arStatus.Status.State
+		}
 	}
 
-	if apiStatus.Status.APIRuleStatus.Code == "ERROR" {
+	if code == "ERROR" || code == "Error" {
```

Comment:
We could also think about making it obvious why we have the different errors and where it comes from, e.g. replacing the strings with variables.

```go
const v1beta1Error = "ERROR"
const v2alpha1Error = "Error"
```

Reply:
Again, I think whole code has to be refactored. This is just bug fix for the time being. I am okay with your proposal.